open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidationUtils.java

```diff
@@ -0,0 +1,132 @@
+import com.wso2.openbanking.accelerator.consent.extensions.manage.model.ConsentManageData;
+import com.wso2.openbanking.accelerator.consent.mgt.service.ConsentCoreService;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * Class to hold idempotency validation utils.
+ */
+public class IdempotencyValidationUtils {
+
+    private static final Log log = LogFactory.getLog(IdempotencyValidationUtils.class);
+    private static final OpenBankingConfigParser parser = OpenBankingConfigParser.getInstance();
+    private static final ConsentCoreService consentCoreService = ConsentExtensionsDataHolder.getInstance()
+            .getConsentCoreService();
+
+    /**
+     * Method to check whether the required parameters for idempotency validation are present.
+     *
+     * @param consentManageData Consent Manage Data
+     * @return Whether the required parameters are present
+     */
+    static boolean isMandatoryParamsPresent(ConsentManageData consentManageData, String idempotencyKeyName) {
+        String idempotencyKeyValue = consentManageData.getHeaders().get(idempotencyKeyName);
+        return idempotencyKeyValue != null && consentManageData.getPayload() != null &&
+                consentManageData.getClientId() != null;
+    }
+
+    /**
+     * Method to check whether Idempotency handling is required.
+     *
+     * @return True if idempotency is required, else False.
+     */
+    static boolean isIdempotencyEnabledFromConfig() {
+        return parser.isIdempotencyValidationEnabled();
+    }
+
+    /**
+     * Method to retrieve the consent ids that have the idempotency key name and value as attribute.
+     *
+     * @param idempotencyKeyName     Idempotency Key Name
+     * @param idempotencyKeyValue    Idempotency Key Value
+     * @return   List of consent ids
+     */
+    static ArrayList<String> getConsentIdsFromIdempotencyKey(String idempotencyKeyName,
+                                                             String idempotencyKeyValue) {
+        try {
+            return consentCoreService.getConsentIdByConsentAttributeNameAndValue(
+                    idempotencyKeyName, idempotencyKeyValue);
+        } catch (ConsentManagementException e) {
+            log.debug("No consent ids found for the idempotency key value");
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Method to compare the client ID sent in the request and client id retrieved from the database.
+     *
+     * @param requestClientID     Client ID sent in the request
+     * @param dbClientId          client ID retrieved from the database
+     * @return   Whether JSON client Ids are equal
+     */
+    static boolean isClientIdsMatching(String requestClientID, String dbClientId) {
```

Comment:
Why are we using a separate method to do the string matching?

Reply:
Added null checks as well inside the method